contract_api/registry.py

```diff
@@ -0,0 +1,273 @@
+                "SELECT G.*, E.* FROM service_group G INNER JOIN service_endpoint E ON G.group_id = E.group_id WHERE "
+                "G.org_id = %s AND G.service_id = %s ", [org_id, service_id])
+            self.obj_utils.clean(group_data)
+            groups = {}
+            for rec in group_data:
+                group_id = rec['group_id']
+                if group_id not in groups.keys():
+                    groups = {group_id: {"group_id": rec['group_id'],
+                                         "group_name": rec['group_name'],
+                                         "pricing": json.loads(rec['pricing']),
+                                         "endpoints": []
+                                         }
+                              }
+                    groups[group_id]['endpoints'].append({"endpoint": rec['endpoint'], "is_available":
+                                                          rec['is_available'], "last_check_timestamp": rec["last_check_timestamp"]})
+            return list(groups.values())
+        except Exception as e:
+            print(repr(e))
+            raise e
+
+    def _get_is_available_service(self):
+        try:
+            available_service = []
+            srvc_st_dta = self.repo.execute(
+                "SELECT DISTINCT org_id, service_id  FROM service_endpoint WHERE is_available = 1")
+            for rec in srvc_st_dta:
+                available_service.append((rec['org_id'], rec['service_id']), )
+            return available_service
+        except Exception as err:
+            print(repr(err))
+
+    def _filter_condition_to__query(self, filter_condition):
+        value = []
+        if filter_condition.attr in ["org_id", "service_id"]:
+            filter_condition.attr = "M." + filter_condition.attr
+        if filter_condition.operator == "=":
+            value = filter_condition.value
+            return '%s %s"%s"' % (filter_condition.attr, filter_condition.operator, "%s"), value
+        if filter_condition.operator == "IN":
+            value = filter_condition.value
+            return '%s %s %s' % (filter_condition.attr, filter_condition.operator, "(" + (("%s,")*len(value))[:-1] + ")"), value
+        if filter_condition.operator == "BETWEEN":
+            value = filter_condition.value
+            return '%s %s %s AND %s' % (filter_condition.attr, filter_condition.operator, "%s", "%s"), value
+
+    def _filters_to_query(self, filter_json):
+        query = ""
+        filters = []
+        values = []
+        for filter in filter_json:
+            filters.append(Filter(filter))
+        for filter in filters:
+            query += "("
+            for filter_condition in filter.get_filter().get("filter"):
+                sub_query, value = self.filter_condition_to__query(
+                    filter_condition)
+                values += value
+                query = query + "(" + sub_query + ") AND "
+            if query.endswith(" AND "):
```

Comment:
Why are we doing query[: -5 ]? The string building logic should not append AND by default rather pre-pend AND if we have multiple conditions.  Then we do not need to do such positional parsing